Mathlib/Analysis/Convex/Radon.lean

```diff
@@ -3,27 +3,37 @@ Copyright (c) 2023 Vasily Nesterov. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Vasily Nesterov
 -/
+import Mathlib.Topology.Separation
 import Mathlib.Analysis.Convex.Combination
-import Mathlib.Tactic.Linarith
+import Mathlib.LinearAlgebra.AffineSpace.FiniteDimensional
 
 /-!
 # Radon's theorem on convex sets
 
 Radon's theorem states that any affine dependent set can be partitioned into two sets whose convex
 hulls intersect.
 
+As a corollary, we prove Helly's theorem, which is a basic result in discrete geometry on the
+intersection of convex sets. Let `X₁, ⋯, Xₙ` be a finite family of convex sets in `ℝᵈ` with
+`n ≥ d + 1`. The theorem states that if any `d + 1` sets from this family intersect nontrivially,
```

Comment:
You don't need `n ≥ d + 1` at all, right?

Reply:
I am not sure here. In the [classical form](https://en.wikipedia.org/wiki/Helly%27s_theorem), Helly's theorem requires `n ≥ d + 1` and the intersection of any (exactly) `d + 1` sets to be nonempty, while I drop the `n ≥ d + 1` assumption but assume that the intersection of any collection of *no more* than `d + 1` intersects nontrivially. The classical form follows from the proved one trivially, but maybe we should have the former as `helly_theorem` and the latter as its version.

Reply:
Sure, I guess that's minutiae

Reply:
I decided to add the classical variation only for the main theorem. Otherwise, we would have 8 Helly's theorems: (families/sets) × (finite/compact) × (classical form/the other form).

Reply:
Is that bad? After all that's what API is for.

See eg `Algebra.Order.Rearrangement` for an example of many copies of one statement